c10/core/DispatchKeySet.h

```diff
@@ -757,6 +757,9 @@ constexpr auto functorch_transforms_ks = DispatchKeySet(
      DispatchKey::VmapMode,
      DispatchKey::FuncTorchGradWrapper});
 
+constexpr auto functorch_batched_ks =
+    DispatchKeySet({DispatchKey::FuncTorchBatched});
```

Comment:
Not necessary for this PR - as we add alias keys and reasoning about them becomes more complicated, it would be a good idea to add tests that some of our alias sets don't overlap. The current state is:

All of these overlap in some capacity (and so their ordering in OperatorEntry.cpp is important)
```
CompositeExplicitAutograd
CompositeExplicitAutogradNonFunctional
CompositeImplicitAutograd
CompositeImplicitAutogradNestedTensor
Autograd
```

And these alias sets are disjoint from every other alias set (so their precedence / ordering doesn't really matter)
```
CompositeImplicitBatched
CompositeImplicitAutogradNestedTensor
```

Reply:
@bdhirsh I'm happy to extend this PR: https://github.com/pytorch/pytorch/pull/89465 do you want to discuss there?

Reply:
I was just thinking out loud :) A test for this would probably be better in C++ than in python, since dispatch keys are exposed in python with more indirection (and for example, I don't think we expose what runtime keys are in an alias key to python).

The more important think is that we document that behavior, so we probably want something like:

(1) A comment next to those two alias keys in `DispatchKey.h`, mentioning that they are disjoint from all the other alias keys, and so their precedence order doesn't matter. (`Note: [Disjoint Alias Keyset]`)

(2) A reference to that comment in `OperatorEntry.cpp` (`See Note: [Disjoint AliasKeyset]`) mentioning that the order for those two alias keys doesn't matter

(3) (optional) a test, maybe in `c10/test/core/DispatchKeySet_test.cpp` that checks that `getRuntimeDispatchKeySet(FuncTorchBatchedDecomposition)` is disjoint from the union of all other alias keysets

Reply:
comments added in https://github.com/pytorch/pytorch/pull/88771/commits/37893e97615342c26513da54425ad884e15441ba

Reply:
thanks!